### ISSUE TITLE
Async insert command

### ISSUE THREAD

**Issue by Jurgen-Aquilina** [ArcadeData/arcadedb#294]
## Expected behavior
Added vertex of type V, with id set to 1

## Actual behavior
Added vertex of type V with no property values. Adding an error callback doesn't output any error. This is only an issue with positional/named arguments while specifying the values directly in the SQL query works. 

## Steps to reproduce
```
   val parameters = mutableMapOf<String, Any>()
    parameters["id"] = "1"
    db.async().command("sql", "insert into V set id = :id", null, parameters)
```

**Comment by lvca** [comment]
@Jurgen-Aquilina could you please write the piece of code you use to create the record asynchronously?

**Comment by Jurgen-Aquilina** [comment]
@lvca not sure I understand. It is within the "steps to reproduce" section above.

**Comment by lvca** [comment]
@Jurgen-Aquilina sorry I thought the error was with calling `database.async().createRecord()` API that, by the way, is faster than executing the command.

Example:

```java
database.async().createRecord(sourceVertex, newDocument -> {
});
```

Abou the command I tried to reproduce it without success:

```java

  @Test
  public void testCommandFetchVarargParamsNoCallback() {
    database.begin();
    try {
      database.async().command("sql", "insert into " + TYPE_NAME + " set id = :id", null, Integer.MAX_VALUE);

      database.async().waitCompletion(5_000);

      final IndexCursor resultSet = database.lookupByKey(TYPE_NAME, "id", Integer.MAX_VALUE);

      Assertions.assertTrue(resultSet.hasNext());
      final Document record = resultSet.next().asDocument();
      Assertions.assertEquals(Integer.MAX_VALUE, record.get("id"));

    } finally {
      database.commit();
    }
  }
```

Could you please check how this code is different from yours?

**Comment by Jurgen-Aquilina** [comment]
@lvca the only difference is that you are using the method with "Object..." parameter while I am using the method Map<String, Object> parameters. Essentially,  the command line should be as follows:
```
      database.async().command("sql", "insert into " + TYPE_NAME + " set id = :id", null, parameters);
```

I also tried using the createRecord() API, but after loading a large number of records, some vertices were lost (despite the callback indicating that the record was saved without errors). I will try to reproduce the issue in a test case and reply here.

**Comment by Jurgen-Aquilina** [comment]
@lvca as mentioned above the createRecord() doesn't create all records. Below is test code reproducing the issue 

```
for (i in 0..250000) {
	val vertex = database.newVertex("V")
	vertex.set("id", i.toString())

	database.async().createRecord(vertex) {}
}

database.async().waitCompletion()

database.schema.createTypeIndex(Schema.INDEX_TYPE.LSM_TREE, true, "V", "id")

var counter = 0

for (i in 0..250000) {
	database.async().query("sql", "SELECT FROM V WHERE id = ?", object : AsyncResultsetCallback {
		override fun onStart(resultset: ResultSet?) {}

		override fun onNext(result: Result?): Boolean { return true }

		override fun onComplete() { counter++ }

		override fun onError(exception: Exception?) {}
	}, i)
}

database.async().waitCompletion()

println(counter)
```

This outputs 249412 instead of 250001

**Comment by Jurgen-Aquilina** [comment]
In case this helps, running a similar experiment with COUNT(*) returns the correct number 250001 indicating that the vertices are created, but with the wrong id set.

```
    for (i in 0..250000) {
        val vertex = database.newVertex("GENE")
        vertex.set("id", i.toString())

        database.async().createRecord(vertex) {}
    }

    database.async().waitCompletion()

    val result = database.query("sql", "SELECT COUNT(*) FROM GENE", null)
    val count = result.next().getProperty<Long>("COUNT(*)")

    println(count)
```

**Comment by Jurgen-Aquilina** [comment]
In case this helps, running this experiment instead returns the correct count of 250001 indicating that all the records are created but with the wrong id set.

```

for (i in 0..250000) {
	val vertex = database.newVertex("V")
	vertex.set("id", i.toString())

	database.async().createRecord(vertex) {}
}

database.async().waitCompletion()

val result = database.query("sql", "SELECT COUNT(*) FROM V", null)
val count = result.next().getProperty<Long>("COUNT(*)")

println(count)

```

**Comment by lvca** [comment]
From looking at your code the insert works (the count returns the correct number of records), but you're executing 250K queries with one counter incremented that is not atomic. Try using `var counter = new AtomicInteger()` and increment it with `counter.incrementAndGet()`

**Comment by Jurgen-Aquilina** [comment]
@lvca you're right. Apologies for that. I also noticed the same issue with edge creation though, so I will investigate further and get back to you.

**Comment by lvca** [comment]
NP.

**Comment by Jurgen-Aquilina** [comment]
@lvca I'm seeing that you closed the issue. To clarify there was also an issue with the following, not setting the vertex properties which I'm not sure whether it was fixed yet:
```
 database.async().command("sql", "insert into " + TYPE_NAME + " set id = :id", null, parameters);
```

The above is erroneous only when parameters is a Map instead of Object...

**Comment by lvca** [comment]
Hi Jurgen, I fixed this issue a couple of days ago. Are you using the latest snapshot?

**Comment by Jurgen-Aquilina** [comment]
@lvca I mentioned it as a reminder as I didn't notice that you fixed it. I'm currently on 22.1.2, but I can test the snapshot. Is it accessible through maven or do I need to add the JAR manually? If it is accessible through maven, can you link the repository please, as I cannot find it?

**Comment by lvca** [comment]
Closing this issue to release 22.1.3. In case you still experience the same problem, please comment on this issue. Thanks, @Jurgen-Aquilina.

**Comment by lvca** [comment]
Released 22.1.3
